minari/utils.py

```diff
@@ -494,11 +508,24 @@ def create_dataset_from_buffers(
         env_spec = env
     elif isinstance(env, gym.Env):
         env_spec = env.spec
+    elif env is None:
+        env_spec = None
     else:
-        raise ValueError("The `env` argument must be of types str|EnvSpec|gym.Env")
+        raise ValueError("The `env` argument must be of types str|EnvSpec|gym.Env|None")
+
+    if isinstance(env, (str, EnvSpec, type(None))) and (observation_space is None or action_space is None):
+        raise ValueError("Both observation space and action space must be provided, if env is str|EnvSpec|None")
```

Comment:
I'm assuming here that action_space and observation_space cannot be extracted unless it is of type gym.Env

Reply:
So, if it is a str or EnvSpec, there would be 2 steps to extract the action_space and observation_space? 

1. recover env as gym.Env with gym.make; 
2. recover action_space and observation_space from this gym.Env.

Is this correct?

Reply:
Yes, exactly

Reply:
implemented the above logic